src/main/java/io/prestosql/plugin/tiledb/TileDBSplitManager.java

```diff
@@ -280,11 +282,16 @@ else if (leftOvers > 0) {
                     LOG.warn("Low > high while setting ranges."); //TODO investigate why
                     return ranges;
                 }
-                if (low != high || lowBound == upperBound) {
-                    ranges.add(new Range(
-                            new Marker(range.getType(), Optional.of(nativeValueToBlock(range.getType(), low)), lowBound),
-                            new Marker(range.getType(), Optional.of(nativeValueToBlock(range.getType(), high)), upperBound)));
+                else if (low == high) {
```

Comment:
This condition is the if statement passes all tests. 
```
                if (low != high || highInclusive == lowerInclusive) {
                    ranges.add(range(
                            range.getType(),
                            low,
                            lowerInclusive,
                            high,
                            highInclusive));
                }
```

Reply:
Confirmed, this passed. Changed accordingly.